packages/edit-post/src/components/header/template-title/delete-template.js

```diff
@@ -42,53 +48,57 @@ export default function DeleteTemplate() {
 		templateTitle = template.title;
 	}
 
+	const onDelete = () => {
+		clearSelectedBlock();
+		setIsEditingTemplate( false );
+
+		editPost( {
+			template: '',
+		} );
+		const settings = getEditorSettings();
+		const newAvailableTemplates = pickBy(
+			settings.availableTemplates,
+			( _title, id ) => {
+				return id !== template.slug;
+			}
+		);
+		updateEditorSettings( {
+			...settings,
+			availableTemplates: newAvailableTemplates,
+		} );
+		deleteEntityRecord( 'postType', 'wp_template', template.id );
+	};
```

Comment:
Should we close the dialog when confirming it by adding a `setShowConfirmDialog( false )` at the end of this callback? The current implementation (using the native confirm()) seems to behave that way. (see also [related comment](https://github.com/WordPress/gutenberg/pull/37491#discussion_r780805958))

Reply:
It looks like the parent component being unmounted is automatically unmounting the `ConfirmDialog`. I thought it was a bug, but confirmed that it's just the aforementioned side effect. One could argue `setShowConfirmDialog( false )` is not needed now, but we don't know if the behavior will change in the future, so best to add it at the bottom of this callback.

Reply:
I agree with @fullofcaffeine — we should `setShowConfirmDialog( false )` here regardless

Reply:
Sounds good, I'll update the PR either later today or early on Monday!

Reply:
Done in 4dcd72d